### PR TITLE
Add Environment variable for terms and conditions page URL #580

### PR DESCRIPTION
**Pull request by ShubhamSingh-S34**
I have added the `NEXT_PUBLIC_TERMS_AND_CONDITIONS_URL="/terms-conditions"` in .env.example file and made respective changes in Join.tsx and JoinWithInvitation.tsx file

### CODE REVIEW

File: components/auth/Join.tsx [hunk 1]
```diff
@@ -104,7 +104,7 @@ const Join = () => {
             <span className="checkbox-toggle"></span>
           </div>
           <label className="label">
-            <span className="label-text">Agree to <Link href='/terms-condition' className='text-primary' target="_blank">Terms and conditions</Link></span>
+            <span className="label-text">Agree to <Link href={`${process.env.NEXT_PUBLIC_TERMS_AND_CONDITIONS_URL}`} className='text-primary' target="_blank">Terms and conditions</Link></span>
```
Comment (devkiran) [inline components/auth/Join.tsx:107]: Can we hide the terms and condition checkbox if `NEXT_PUBLIC_TERMS_AND_CONDITIONS_URL` is not set? This way it won't break for existing users of the starter kit.

**Review by devkiran — CHANGES_REQUESTED** [review]

**Comment by ShubhamSingh-S34** [comment]
Done

File: components/auth/Join.tsx [hunk 2]
```diff
@@ -95,18 +95,19 @@ const Join = () => {
...
-          <div className="space-x-2">
-            <Checkbox type="checkbox" className='checkbox checkbox-primary checkbox-xs' onChange={(e) => {
-              formik.setFieldValue('agreeToTerms', e.target.checked);
-            }} />
-            <span className="checkbox-toggle"></span>
-          </div>
-          <label className="label">
-            <span className="label-text">Agree to <Link href='/terms-condition' className='text-primary' target="_blank">Terms and conditions</Link></span>
-          </label>
-        </div>
```
Comment (devkiran) [inline components/auth/Join.tsx:109]: You have to conditionally add this validation if the env variable is set

```
      agreeToTerms: Yup.boolean().oneOf(
        [true],
        'You must agree to the Terms and Conditions.'
      ),
```

File: .env.example [hunk 3]
```diff
@@ -54,4 +54,7 @@ OTEL_EXPORTER_OTLP_METRICS_ENDPOINT=
 OTEL_EXPORTER_OTLP_METRICS_HEADERS=
 OTEL_EXPORTER_OTLP_METRICS_PROTOCOL=grpc
 OTEL_EXPORTER_DEBUG=true
-OTEL_PREFIX=boxyhq.saas
\ No newline at end of file
+OTEL_PREFIX=boxyhq.saas
+
+
```
Comment (devkiran) [inline .env.example:59]:
```suggestion

```

**Review by devkiran — CHANGES_REQUESTED** [review]

File: .env.example (hunk 3 above)
Comment (ShubhamSingh-S34) [reply .env.example:59]: Just a small query..
is NEXT_PUBLIC_TERMS_AND_CONDITIONS_URL not defined or just an empty string?

Comment (devkiran) [reply .env.example:59]: Both cases. You can do `if(process.env.NEXT_PUBLIC_TERMS_AND_CONDITIONS_URL)`.

**Comment by ShubhamSingh-S34** [comment]
Done

File: components/auth/Join.tsx [hunk 4]
```diff
@@ -27,7 +27,9 @@ const Join = () => {
       email: Yup.string().required().email(),
       password: Yup.string().required().min(8),
       team: Yup.string().required().min(3),
-      agreeToTerms: Yup.boolean().oneOf([true], 'You must agree to the Terms and Conditions.'),
+      agreeToTerms: process.env.NEXT_PUBLIC_TERMS_AND_CONDITIONS_URL
+        ? Yup.boolean().oneOf([true], 'You must agree to the Terms and Conditions.')
```
Comment (ShubhamSingh-S34) [inline components/auth/Join.tsx:31]: Adding conditional statement here.
if NEXT_PUBLIC_TERMS_AND_CONDITIONS_URL is defined then agreeToTerms has to be true
else agreeToTerms can be a boolean which is initially defined as false

File: components/auth/Join.tsx [hunk 5]
```diff
@@ -95,18 +97,19 @@ const Join = () => {
...
-            <Checkbox type="checkbox" className='checkbox checkbox-primary checkbox-xs' onChange={(e) => {
-              formik.setFieldValue('agreeToTerms', e.target.checked);
-            }} />
-            <span className="checkbox-toggle"></span>
-          </div>
-          <label className="label">
-            <span className="label-text">Agree to <Link href='/terms-condition' className='text-primary' target="_blank">Terms and conditions</Link></span>
-          </label>
-        </div>
+        {process.env.NEXT_PUBLIC_TERMS_AND_CONDITIONS_URL?.length !== 0 &&
```
Comment (devkiran) [inline components/auth/Join.tsx:100]:
```suggestion
        {process.env.NEXT_PUBLIC_TERMS_AND_CONDITIONS_URL &&
```
When NEXT_PUBLIC_TERMS_AND_CONDITIONS_URL is not set (missing), the value is undefined.

File: components/auth/JoinWithInvitation.tsx [hunk 6]
```diff
@@ -95,17 +97,20 @@ const JoinWithInvitation = ({
         error={formik.touched.password ? formik.errors.password : undefined}
         onChange={formik.handleChange}
       />
-      <div className="form-control flex  flex-row items-center">
-        <div className="space-x-2">
-          <Checkbox type="checkbox" className='checkbox checkbox-primary checkbox-xs' onChange={(e) => {
-            formik.setFieldValue('agreeToTerms', e.target.checked);
-          }} />
-          <span className="checkbox-toggle"></span>
+      {process.env.NEXT_PUBLIC_TERMS_AND_CONDITIONS_URL?.length !== 0 &&
```
Comment (devkiran) [inline components/auth/JoinWithInvitation.tsx:100]:
```suggestion
      {process.env.NEXT_PUBLIC_TERMS_AND_CONDITIONS_URL &&
```

**Review by devkiran — CHANGES_REQUESTED** [review]

**Review by devkiran — APPROVED** [review]

**Comment by NathanTarbert** [comment]
Thank you @ShubhamSingh-S34 for your valued contribution!

**Pull request merged**